File: v2.x/Examples/SciChartDemo/ShareChartExample/ShareChartExample/CustomModifierControlPanel.m
//
//  CustomModifierControlPanel.m
//  SciChartDemo
//
//  Created by Admin on 29/08/16.
//  Copyright © 2016 ABT. All rights reserved.
//

#import "CustomModifierControlPanel.h"

@interface CustomModifierControlPanel()

@property (weak, nonatomic) IBOutlet UILabel *infoLabel;

@end

@implementation CustomModifierControlPanel

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

-(void)setText:(NSString *)text {
    _infoLabel.text = [text copy];
}

- (IBAction)prevButtonClick:(id)sender {
    if (_onPrevClicked) _onPrevClicked();
}

- (IBAction)nextButtonClick:(id)sender {
    if (_onNextClicked) _onNextClicked();
}

- (IBAction)clearButtonClick:(id)sender {
    if (_onClearClicked) _onClearClicked();
}

@end

File: v2.x/Examples/SciChartDemo/ShareChartSwiftExample/ShareChartSwiftExample/CustomModifierControlPanel.swift
//
//  CustomModifierControlPanel.swift
//  SciChartSwiftDemo
//
//  Created by Admin on 20/09/16.
//  Copyright © 2016 SciChart Ltd. All rights reserved.
//

import Foundation
import UIKit

typealias CustomModifierControlPanelCallback = () -> Void

open class CustomModifierControlPanel : UIView {
    
    var onClearClicked : CustomModifierControlPanelCallback!
    var onPrevClicked : CustomModifierControlPanelCallback!
    var onNextClicked : CustomModifierControlPanelCallback!
    
    @IBOutlet weak var infoLabel: UILabel!
    
    @IBAction func prevClick(_ sender: AnyObject) {
        if (onPrevClicked != nil) {
            onPrevClicked()
        }
    }
    
    @IBAction func nextClick(_ sender: AnyObject) {
        if (onNextClicked != nil) {
            onNextClicked()
        }
    }
    
    @IBAction func clearClick(_ sender: AnyObject) {
        if (onClearClicked != nil) {
            onClearClicked()
        }
    }
    
    open func setText(_ text : String ) {
        infoLabel.text = text
    }
    
}
